### PR TITLE
chore(deps): Updated dependencies in the test project

### PR DESCRIPTION
**Pull request by nikcio**
This updates the packages found in the test project to newer versions.

It includes:

- Framework update to .Net 6.0
- Microsoft.Extensions.Hosting 6.0.1
- Microsoft.Extensions.Logging 6.0.0
- Microsoft.Extensions.Logging.Console 6.0.0
- Microsoft.NET.Test.Sdk 17.3.2
- Moq 4.18.2
- NUnit 3.13.3
- Nunit3TestAdapter 4.2.1
- System.Configuration.ConfigurationManager 6.0.1
- Switch from Microsoft.Azure.Storage.Blob to Azure.Storage.Blob 12.13.1

### CODE REVIEW

**Comment by Shazwazza** [comment]
Hi, thanks for the effort here. As per previous msg, the branch for v3 is https://github.com/Shazwazza/Examine/tree/release/3.0. I'll go ahead and update the default branch for this repo to that branch to avoid more confusion. If you want to retarget the dependency changes to that branch, that would be great. 

These changes in the test project are ok since we are not releasing this project as a public Nuget. However, please keep in mind we cannot introduce any breaking changes to Examine 3.x for the public consuming packages. If we want to build for both .NET Standard and .NET 6 then we need to multi-target the output.

**Comment by nikcio** [comment]
This has now been rebased to the correct branch and the target branch has been changed.

File: src/Examine.Test/Examine.Test.csproj [hunk 1]
```diff
@@ -11,7 +11,7 @@
     </SccProvider>
   </PropertyGroup>
   <PropertyGroup>
-    <TargetFramework>net5.0</TargetFramework>
+    <TargetFramework>net6.0</TargetFramework>
```
Comment (Shazwazza) [inline src/Examine.Test/Examine.Test.csproj:14]: Can we multi-target this so there are no breaking changes?

File: src/Examine.Test/Examine.Test.csproj [hunk 2]
```diff
@@ -53,23 +53,21 @@
...
     </PackageReference>
-    <PackageReference Include="Microsoft.Azure.Storage.Blob">
-      <Version>11.2.2</Version>
-    </PackageReference>
-    <PackageReference Include="Microsoft.Extensions.Hosting" Version="5.0.0" />
-    <PackageReference Include="Microsoft.Extensions.Logging" Version="5.0.0" />
-    <PackageReference Include="Microsoft.Extensions.Logging.Console" Version="5.0.0" />
-    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.9.4" />
-    <PackageReference Include="Moq" Version="4.16.1" />
+    <PackageReference Include="Microsoft.Extensions.Hosting" Version="6.0.1" />
```
Comment (Shazwazza) [inline src/Examine.Test/Examine.Test.csproj:60]: If we multi-target, then I think we need to keep the v5 versions around for the v5 output?

**Review by Shazwazza — CHANGES_REQUESTED** [review]

File: src/Examine.Test/Examine.Test.csproj (hunk 1 above)
Comment (nikcio) [reply src/Examine.Test/Examine.Test.csproj]: I've now added multi targeting to the test project. This means that the tests will run on both .NET 5 and .NET 6 when running `dotnet test`. Do note that the Visual Studio test explorer only runs the tests on one framework (The one placed first). Source: https://stackoverflow.com/questions/44477840/how-to-properly-unit-test-a-net-project-with-multiple-target-frameworks-given

I've also added .NET 6 support to the build workflow to allow the GitHub action to run the tests in .NET 6.

File: src/Examine.Test/Examine.Test.csproj (hunk 2 above)
Comment (nikcio) [reply src/Examine.Test/Examine.Test.csproj:60]: All the changed packages have multi targeting and will work on both .NET 5 and .NET 6. Example: https://www.nuget.org/packages/Microsoft.Extensions.Hosting

Therefore, I don't think it makes much sense to keep two different versions. All the packages are just for testing anyways and will therefore not affect users using Examine.

**Pull request merged**